### ISSUE TITLE
Error when running showWaterfall()

### ISSUE THREAD

**Issue by msmith01** [AppliedDataSciencePartners/xgboostExplainer#22]
I receive an error when running xgboostExplainer. The error occurs when I run the showWaterfall() command.

```
library(xgboost)
library(xgboostExplainer)
data(agaricus.train, package='xgboost')
data(agaricus.test, package='xgboost')
train <- agaricus.train
test <- agaricus.test

xgb.train.data <- xgb.DMatrix(train$data, label = train$label)
xgb.test.data <- xgb.DMatrix(test$data, label = test$label)
param <- list(objective = "binary:logistic")

model.cv <- xgb.cv(param = param,
                   data = xgb.train.data,
                   nrounds = 500,
                   early_stopping_rounds = 10,
                   nfold = 3)

model.cv$best_ntreelimit

xgb.model <- xgboost(param =param,  data = xgb.train.data, nrounds = 10)

explainer <- buildExplainer(xgb.model, xgb.train.data, type="binary", base_score = 0.5, n_first_tree = 5)

pred.breakdown = explainPredictions(xgb.model,
                                    explainer,
                                    xgb.test.data)

showWaterfall(xgb.model,
              explained,
              xgb.test.data, test$data,  2, type = "binary")
```

ERROR:

```
Extracting the breakdown of each prediction...
  |=================================================| 100%

DONE!

Actual:  1
Prediction:  NA
Weight:  NA
Breakdown
                       intercept 
                              NA 
                  cap-shape=bell 
                              NA 
               cap-shape=conical 
                              NA 
                cap-shape=convex 
                              NA 
                  cap-shape=flat 
                              NA 
               cap-shape=knobbed 
                              NA 
                cap-shape=sunken 
                              NA 
             cap-surface=fibrous 
                              NA 
             cap-surface=grooves 
                              NA 
               cap-surface=scaly 
                              NA 
              cap-surface=smooth 
                              NA 
                 cap-color=brown 
                              NA 
                  cap-color=buff 
                              NA 
              cap-color=cinnamon 
                              NA 
                  cap-color=gray 
                              NA 
                 cap-color=green 
                              NA 
                  cap-color=pink 
                              NA 
                cap-color=purple 
                              NA 
                   cap-color=red 
                              NA 
                 cap-color=white 
                              NA 
                cap-color=yellow 
                              NA 
                bruises?=bruises 
                              NA 
                     bruises?=no 
                              NA 
                     odor=almond 
                              NA 
                      odor=anise 
                              NA 
                   odor=creosote 
                              NA 
                      odor=fishy 
                              NA 
                       odor=foul 
                              NA 
                      odor=musty 
                              NA 
                       odor=none 
                              NA 
                    odor=pungent 
                              NA 
                      odor=spicy 
                              NA 
        gill-attachment=attached 
                              NA 
      gill-attachment=descending 
                              NA 
            gill-attachment=free 
                              NA 
         gill-attachment=notched 
                              NA 
              gill-spacing=close 
                              NA 
            gill-spacing=crowded 
                              NA 
            gill-spacing=distant 
                              NA 
                 gill-size=broad 
                              NA 
                gill-size=narrow 
                              NA 
                gill-color=black 
                              NA 
                gill-color=brown 
                              NA 
                 gill-color=buff 
                              NA 
            gill-color=chocolate 
                              NA 
                 gill-color=gray 
                              NA 
                gill-color=green 
                              NA 
               gill-color=orange 
                              NA 
                 gill-color=pink 
                              NA 
               gill-color=purple 
                              NA 
                  gill-color=red 
                              NA 
                gill-color=white 
                              NA 
               gill-color=yellow 
                              NA 
           stalk-shape=enlarging 
                              NA 
            stalk-shape=tapering 
                              NA 
              stalk-root=bulbous 
                              NA 
                 stalk-root=club 
                              NA 
                  stalk-root=cup 
                              NA 
                stalk-root=equal 
                              NA 
          stalk-root=rhizomorphs 
                              NA 
               stalk-root=rooted 
                              NA 
              stalk-root=missing 
                              NA 
stalk-surface-above-ring=fibrous 
                              NA 
  stalk-surface-above-ring=scaly 
                              NA 
  stalk-surface-above-ring=silky 
                              NA 
 stalk-surface-above-ring=smooth 
                              NA 
stalk-surface-below-ring=fibrous 
                              NA 
  stalk-surface-below-ring=scaly 
                              NA 
  stalk-surface-below-ring=silky 
                              NA 
 stalk-surface-below-ring=smooth 
                              NA 
    stalk-color-above-ring=brown 
                              NA 
     stalk-color-above-ring=buff 
                              NA 
 stalk-color-above-ring=cinnamon 
                              NA 
     stalk-color-above-ring=gray 
                              NA 
   stalk-color-above-ring=orange 
                              NA 
     stalk-color-above-ring=pink 
                              NA 
      stalk-color-above-ring=red 
                              NA 
    stalk-color-above-ring=white 
                              NA 
   stalk-color-above-ring=yellow 
                              NA 
    stalk-color-below-ring=brown 
                              NA 
     stalk-color-below-ring=buff 
                              NA 
 stalk-color-below-ring=cinnamon 
                              NA 
     stalk-color-below-ring=gray 
                              NA 
   stalk-color-below-ring=orange 
                              NA 
     stalk-color-below-ring=pink 
                              NA 
      stalk-color-below-ring=red 
                              NA 
    stalk-color-below-ring=white 
                              NA 
   stalk-color-below-ring=yellow 
                              NA 
               veil-type=partial 
                              NA 
             veil-type=universal 
                              NA 
                veil-color=brown 
                              NA 
               veil-color=orange 
                              NA 
                veil-color=white 
                              NA 
               veil-color=yellow 
                              NA 
                ring-number=none 
                              NA 
                 ring-number=one 
                              NA 
                 ring-number=two 
                              NA 
              ring-type=cobwebby 
                              NA 
            ring-type=evanescent 
                              NA 
               ring-type=flaring 
                              NA 
                 ring-type=large 
                              NA 
                  ring-type=none 
                              NA 
               ring-type=pendant 
                              NA 
             ring-type=sheathing 
                              NA 
                  ring-type=zone 
                              NA 
         spore-print-color=black 
                              NA 
         spore-print-color=brown 
                              NA 
          spore-print-color=buff 
                              NA 
     spore-print-color=chocolate 
                              NA 
         spore-print-color=green 
                              NA 
        spore-print-color=orange 
                              NA 
        spore-print-color=purple 
                              NA 
         spore-print-color=white 
                              NA 
        spore-print-color=yellow 
                              NA 
             population=abundant 
                              NA 
            population=clustered 
                              NA 
             population=numerous 
                              NA 
            population=scattered 
                              NA 
              population=several 
                              NA 
             population=solitary 
                              NA 
                 habitat=grasses 
                              NA 
                  habitat=leaves 
                              NA 
                 habitat=meadows 
                              NA 
                   habitat=paths 
                              NA 
                   habitat=urban 
                              NA 
                   habitat=waste 
                              NA 
                   habitat=woods 
                              NA 
-3.89182 -3.178054 -2.751535 -2.442347 -2.197225 -1.99243 -1.81529 -1.658228 -1.516347 -1.386294 -1.265666 -1.15268 -1.045969 -0.9444616 -0.8472979 -0.7537718 -0.6632942 -0.5753641 -0.4895482 -0.4054651 -0.3227734 -0.2411621 -0.1603427 -0.08004271 0 0.08004271 0.1603427 0.2411621 0.3227734 0.4054651 0.4895482 0.5753641 0.6632942 0.7537718 0.8472979 0.9444616 1.045969 1.15268 1.265666 1.386294 1.516347 1.658228 1.81529 1.99243 2.197225 2.442347 2.751535 3.178054 3.89182
**Error in if (abs(values[i]) > put_rect_text_outside_when_value_below) { : 
  missing value where TRUE/FALSE needed**
```

**Comment by msmith01** [comment]
To answer my own issue, it has been resolved. If somebody else has this issue, check your `xgboost` package version. `xgboostExplainer` does not work with the development version on github (version `0.6.4.6`. So if you installed your `xgboost` package via `devtools` then it most probably will not work. Make sure the xgboost package is the same as the cran version, at the time of writing version `0.6.4.1`.

**Comment by davidADSP** [comment]
should now work with latest CRAN version of xgboost (1.7)
fixed by 10e61bb194b09ac5ac23f54ab632f310fcf6b728
